File: app/src/main/java/com/oguz/spy/ux/PlayerSetUpScreen.kt
package com.oguz.spy.ux

import androidx.compose.runtime.Composable
import androidx.navigation.NavController
import androidx.compose.foundation.background
import androidx.compose.foundation.layout.*
import androidx.compose.foundation.lazy.LazyColumn
import androidx.compose.foundation.lazy.itemsIndexed
import androidx.compose.foundation.shape.CircleShape
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.filled.ArrowBack
import androidx.compose.material.icons.filled.PlayArrow
import androidx.compose.material3.*
import androidx.compose.runtime.*
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.graphics.Brush
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import com.oguz.spy.ux.components.PlayerCard
import com.oguz.spy.models.CharacterAvatar // YENİ IMPORT

// Player data class'ını güncelleyin (selectedCharacter alanını ekleyin)
data class Player(
    val id: Int,
    var name: String = "",
    var selectedColor: Color = Color.Gray,
    var selectedCharacter: CharacterAvatar? = null // YENİ ALAN
)

@OptIn(ExperimentalMaterial3Api::class)
@Composable
fun PlayerSetupScreen(
    navController: NavController,
    playerCount: Int = 4,
    existingPlayers: List<Player> = emptyList(),
    onBackClick: () -> Unit = { navController.popBackStack() },
    onStartGame: (List<Player>) -> Unit = { navController.navigate("categoryScreen") }
) {
    // Mevcut renkler (aynı kalacak)
    val availableColors = listOf(
        Color(0xFFE91E63), Color(0xFF9C27B0), Color(0xFF3F51B5),
        Color(0xFF2196F3), Color(0xFF00BCD4), Color(0xFF4CAF50),
        Color(0xFF8BC34A), Color(0xFFFFEB3B), Color(0xFFFF9800),
        Color(0xFFFF5722), Color(0xFF795548), Color(0xFF607D8B)
    )

    // Oyuncular listesi (karakter seçimi için güncellenmiş)
    var players by remember {
        mutableStateOf(
            if (existingPlayers.isNotEmpty() && existingPlayers.size >= playerCount) {
                existingPlayers.take(playerCount).mapIndexed { index, existingPlayer ->
                    existingPlayer.copy(id = index + 1)
                }
            } else {
                val existingPlayersToUse = existingPlayers.take(playerCount)
                val missingPlayersCount = playerCount - existingPlayersToUse.size
                val allPlayers = mutableListOf<Player>()

                existingPlayersToUse.forEachIndexed { index, player ->
                    allPlayers.add(player.copy(id = index + 1))
                }

                repeat(missingPlayersCount) { index ->
                    val playerId = existingPlayersToUse.size + index + 1
                    val usedColors = allPlayers.map { it.selectedColor }
                    val usedCharacters = allPlayers.mapNotNull { it.selectedCharacter }

                    val availableColor = availableColors.firstOrNull {
                        !usedColors.contains(it)
                    } ?: availableColors[(playerId - 1) % availableColors.size]

                    val availableCharacter = CharacterAvatar.values().firstOrNull {
                        !usedCharacters.contains(it)
                    }

                    allPlayers.add(
                        Player(
                            id = playerId,
                            name = "Oyuncu $playerId",
                            selectedColor = availableColor,
                            selectedCharacter = availableCharacter
                        )
                    )
                }
                allPlayers
            }
        )
    }

    val isFormValid = players.all { it.name.isNotBlank() } &&
            players.map { it.selectedColor }.distinct().size == players.size

    Box(
        modifier = Modifier
            .fillMaxSize()
            .background(
                Brush.verticalGradient(
                    colors = listOf(
                        Color(0xFFE91E63),
                        Color(0xFF9C27B0),
                        Color(0xFFF44336)
                    )
                )
            )
    ) {
        Column(
            modifier = Modifier
                .fillMaxSize()
                .padding(bottom = 80.dp, top = 20.dp)
        ) {
            // Top Bar (aynı kalacak)
            Row(
                modifier = Modifier
                    .fillMaxWidth()
                    .padding(16.dp),
                verticalAlignment = Alignment.CenterVertically
            ) {
                IconButton(
                    onClick = onBackClick,
                    modifier = Modifier
                        .clip(CircleShape)
                        .background(Color.White.copy(alpha = 0.2f))
                ) {
                    Icon(
                        imageVector = Icons.Default.ArrowBack,
                        contentDescription = "Geri",
                        tint = Color.White
                    )
                }

                Spacer(modifier = Modifier.width(16.dp))

                Column {
                    Text(
                        text = "Oyuncular",
                        fontSize = 24.sp,
                        fontWeight = FontWeight.Bold,
                        color = Color.White
                    )
                    Text(
                        text = "İsim, renk ve karakter seçin", // GÜNCELLENDİ
                        fontSize = 14.sp,
                        color = Color.White.copy(alpha = 0.8f)
                    )
                }
            }

            // Players List - GÜNCELLENDİ
            LazyColumn(
                modifier = Modifier
                    .fillMaxSize()
                    .padding(horizontal = 16.dp),
                verticalArrangement = Arrangement.spacedBy(16.dp)
            ) {
                itemsIndexed(players) { index, player ->
                    PlayerCard(
                        player = player,
                        availableColors = availableColors,
                        usedColors = players.mapNotNull { if (it.id != player.id) it.selectedColor else null },
                        usedCharacters = players.mapNotNull { if (it.id != player.id) it.selectedCharacter else null }, // YENİ
                        onNameChange = { newName ->
                            players = players.map {
                                if (it.id == player.id) it.copy(name = newName) else it
                            }
                        },
                        onColorChange = { newColor ->
                            players = players.map {
                                if (it.id == player.id) it.copy(selectedColor = newColor) else it
                            }
                        },
                        onCharacterChange = { newCharacter -> // YENİ
                            players = players.map {
                                if (it.id == player.id) it.copy(selectedCharacter = newCharacter) else it
                            }
                        }
                    )
                }

                item {
                    Spacer(modifier = Modifier.height(16.dp))
                }
            }
        }

        // Button (aynı kalacak)
        Box(
            modifier = Modifier
                .align(Alignment.BottomCenter)
                .fillMaxWidth()
                .background(
                    Brush.verticalGradient(
                        colors = listOf(
                            Color.Transparent,
                            Color.Black.copy(alpha = 0.3f)
                        )
                    )
                )
                .padding(16.dp)
        ) {
            Button(
                onClick = {
                    players.forEach { player ->
                        println("${player.name} - Renk: ${player.selectedColor} - Karakter: ${player.selectedCharacter}")
                    }
                    onStartGame(players)
                },
                modifier = Modifier
                    .fillMaxWidth()
                    .height(56.dp),
                enabled = isFormValid,
                colors = ButtonDefaults.buttonColors(
                    containerColor = Color.White,
                    disabledContainerColor = Color.White.copy(alpha = 0.5f)
                ),
                shape = RoundedCornerShape(16.dp)
            ) {
                Icon(
                    imageVector = Icons.Default.PlayArrow,
                    contentDescription = null,
                    tint = if (isFormValid) Color(0xFFE91E63) else Color.Gray,
                    modifier = Modifier.size(20.dp)
                )
                Spacer(modifier = Modifier.width(8.dp))
                Text(
                    text = "Kategori Seç",
                    fontSize = 16.sp,
                    fontWeight = FontWeight.Bold,
                    color = if (isFormValid) Color(0xFFE91E63) else Color.Gray
                )
            }
        }
    }
}

File: lib/screens/player_setup_screen.dart
import 'package:flutter/material.dart';
import 'package:flutter_riverpod/flutter_riverpod.dart';
import 'package:go_router/go_router.dart';
import '../models/player.dart';
import '../models/character_avatar.dart';
import '../widgets/player_card.dart';

final playersProvider = StateProvider<List<Player>>((ref) => []);

class PlayerSetupScreen extends ConsumerStatefulWidget {
  final int playerCount;
  final List<Player> existingPlayers;

  const PlayerSetupScreen({
    super.key,
    required this.playerCount,
    this.existingPlayers = const [],
  });

  @override
  ConsumerState<PlayerSetupScreen> createState() => _PlayerSetupScreenState();
}

class _PlayerSetupScreenState extends ConsumerState<PlayerSetupScreen> {
  final List<Color> availableColors = const [
    Color(0xFFE91E63), Color(0xFF9C27B0), Color(0xFF3F51B5),
    Color(0xFF2196F3), Color(0xFF00BCD4), Color(0xFF4CAF50),
    Color(0xFF8BC34A), Color(0xFFFFEB3B), Color(0xFFFF9800),
    Color(0xFFFF5722), Color(0xFF795548), Color(0xFF607D8B),
  ];

  late List<Player> players;

  @override
  void initState() {
    super.initState();
    _initPlayers();
  }

  void _initPlayers() {
    final existing = widget.existingPlayers;
    final count = widget.playerCount;

    if (existing.length >= count) {
      players = existing.take(count).mapIndexed((i, p) => p.copyWith(id: i + 1)).toList();
    } else {
      final result = <Player>[];
      for (int i = 0; i < existing.length; i++) {
        result.add(existing[i].copyWith(id: i + 1));
      }
      for (int i = existing.length; i < count; i++) {
        result.add(Player(id: i + 1, name: 'Oyuncu ${i + 1}'));
      }
      players = result;
    }
  }

  bool get isFormValid => players.every((p) => p.name.trim().isNotEmpty);

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      body: Container(
        decoration: const BoxDecoration(
          gradient: LinearGradient(
            begin: Alignment.topCenter,
            end: Alignment.bottomCenter,
            colors: [
              Color(0xFFE91E63),
              Color(0xFF9C27B0),
              Color(0xFFF44336),
            ],
          ),
        ),
        child: Stack(
          children: [
            Column(
              children: [
                const SizedBox(height: 20),
                // Header
                Padding(
                  padding: const EdgeInsets.symmetric(horizontal: 16),
                  child: Row(
                    children: [
                      GestureDetector(
                        onTap: () => context.pop(),
                        child: Container(
                          width: 40,
                          height: 40,
                          decoration: BoxDecoration(
                            color: Colors.white.withOpacity(0.2),
                            shape: BoxShape.circle,
                          ),
                          child: const Icon(Icons.arrow_back, color: Colors.white),
                        ),
                      ),
                      const SizedBox(width: 16),
                      Column(
                        crossAxisAlignment: CrossAxisAlignment.start,
                        children: [
                          const Text(
                            'Oyuncular',
                            style: TextStyle(
                              fontSize: 24,
                              fontWeight: FontWeight.bold,
                              color: Colors.white,
                            ),
                          ),
                          Text(
                            'İsim, renk ve karakter seçin',
                            style: TextStyle(
                              fontSize: 14,
                              color: Colors.white.withOpacity(0.8),
                            ),
                          ),
                        ],
                      ),
                    ],
                  ),
                ),
                const SizedBox(height: 12),
                // Liste
                Expanded(
                  child: ListView.separated(
                    padding: const EdgeInsets.fromLTRB(16, 0, 16, 100),
                    itemCount: players.length,
                    separatorBuilder: (_, __) => const SizedBox(height: 16),
                    itemBuilder: (context, index) {
                      final player = players[index];
                      return PlayerCard(
                        player: player,
                        availableColors: availableColors,
                        usedColors: players
                            .where((p) => p.id != player.id)
                            .map((p) => p.selectedColor)
                            .whereType<Color>()
                            .toList(),
                        usedCharacters: players
                            .where((p) => p.id != player.id)
                            .map((p) => p.selectedCharacter)
                            .whereType<CharacterAvatar>()
                            .toList(),
                        onNameChange: (name) => setState(() {
                          players = players
                              .map((p) => p.id == player.id ? p.copyWith(name: name) : p)
                              .toList();
                        }),
                        onColorChange: (color) => setState(() {
                          players = players
                              .map((p) => p.id == player.id
                              ? p.copyWith(
                            selectedColor: color,
                            clearColor: color == null,
                          )
                              : p)
                              .toList();
                        }),
                        onCharacterChange: (char) => setState(() {
                          players = players
                              .map((p) => p.id == player.id
                              ? p.copyWith(
                            selectedCharacter: char,
                            clearCharacter: char == null,
                          )
                              : p)
                              .toList();
                        }),
                      );
                    },
                  ),
                ),
              ],
            ),

            // Bottom button
            Positioned(
              bottom: 0,
              left: 0,
              right: 0,
              child: Container(
                decoration: BoxDecoration(
                  gradient: LinearGradient(
                    begin: Alignment.topCenter,
                    end: Alignment.bottomCenter,
                    colors: [
                      Colors.transparent,
                      Colors.black.withOpacity(0.3),
                    ],
                  ),
                ),
                padding: const EdgeInsets.all(16),
                child: SizedBox(
                  height: 56,
                  child: ElevatedButton.icon(
                    onPressed: isFormValid
                        ? () {
                      ref.read(playersProvider.notifier).state = players;
                      context.push('/categoryScreen');
                    }
                        : null,
                    icon: Icon(
                      Icons.play_arrow,
                      color: isFormValid ? const Color(0xFFE91E63) : Colors.grey,
                    ),
                    label: Text(
                      'Kategori Seç',
                      style: TextStyle(
                        fontSize: 16,
                        fontWeight: FontWeight.bold,
                        color: isFormValid ? const Color(0xFFE91E63) : Colors.grey,
                      ),
                    ),
                    style: ElevatedButton.styleFrom(
                      backgroundColor: Colors.white,
                      disabledBackgroundColor: Colors.white.withOpacity(0.5),
                      shape: RoundedRectangleBorder(
                        borderRadius: BorderRadius.circular(16),
                      ),
                    ),
                  ),
                ),
              ),
            ),
          ],
        ),
      ),
    );
  }
}

// Dart'ta mapIndexed extension
extension IndexedIterable<E> on Iterable<E> {
  List<T> mapIndexed<T>(T Function(int index, E element) f) {
    var i = 0;
    return map((e) => f(i++, e)).toList();
  }
}